gobblin-metrics/src/main/java/gobblin/metrics/kafka/KafkaReporter.java

```diff
@@ -85,6 +96,11 @@ protected KafkaReporter(Builder<?> builder) {
     this.config = new ProducerConfig(props);
     this.producer = closer.register(new ProducerCloseable<String, byte[]>(config));
```

Comment:
You probably want to avoid creating the `producer`, `writer`, and `out` if `reportMetrics` is `false`. I suggest make them `Optional`.